### PR TITLE
ECOPROJECT-12: Documentation for the Poison Pill Operator

### PR DESCRIPTION
**Pull request by abhatt-rh**
[TELCODOCS-53](https://issues.redhat.com/browse/TELCODOCS-53): Documentation for the Poison Pill Operator

Applies to OpenShift version 4.8+

Preview:  https://deploy-preview-35974--osdocs.netlify.app/openshift-enterprise/latest/nodes/nodes/eco-poison-pill-operator.html

### CODE REVIEW

**Comment by abhatt-rh** [comment]
Hi @beekhof, @n1r1, and @shellymiron,
Request you to kindly review the Poison Pill Operator Documentation for 4.8 and provide your inputs. Thank you!

File: modules/eco-poison-pill-operator-about.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  name: poison-pill-config
+  namespace: openshift-operators
+  resourceVersion: '31617'
+  uid: 7c9c4502-f847-4deb-ad7d-877551777a49
+spec:
+  safeTimeToAssumeNodeRebootedSeconds: 180 <1>
+  watchdogFilePath: /test/watchdog1 <2>
+----
+
+<1> Specify the timeout duration for the Operator to assume that an unhealthy node is rebooted. The Operator automatically calculates the lower limit for this value. However, for hardware where different nodes have different watchdog timeout, you must change this value to an upper limit.
```
Comment (johnwilkins) [inline modules/eco-poison-pill-operator-about.adoc:54]: s/have different/have a different/

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+----
+
+. Verify that the Poison Pill Operator is up and running:
++
+[source,terminal]
+----
+$ oc get deploy -n poison-pill
+----
+
+. Verify that the `PoisonPillConfig` CR is created:
```
Comment (johnwilkins) [inline modules/eco-poison-pill-operator-installation-cli.adoc:89]: Verify that the Poison Pill Operator created the `PoisonPillConfig` CR:

File: modules/eco-poison-pill-operator-installation-web-console.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// * /machine_management/eco-poison-pill-operator.adoc
+
+[id="installing-poison-pill-operator-using-web-console_{context}"]
+= Installing the Poison Pill Operator using the web console
+
+.Prerequisites
+
+* A cluster installed on bare-metal hardware using installer-provisioned infrastructure (IPI) .
```
Comment (johnwilkins) [inline modules/eco-poison-pill-operator-installation-web-console.adoc:10]: Remove space before period.

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+Issue:: An unhealthy node was not remediated.
+
+Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. To do so, run the following command:
++
+[source,bash,options="nowrap",role="white-space-pre"]
+----
+oc get namespace <machinehealthcheck>
+----
+
+If the `PoisonPillRemediation` CR is not created when the node turned unhealthy, check the logs of the `MachineHealthCheck` controller.
```
Comment (johnwilkins) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:37]: General comment: what is creating the `PoisonPillRemediation` CR? If we can answer that, we can address the passive voice.

**Review by johnwilkins — COMMENTED** [review]
Here are my initial comments...

File: modules/eco-poison-pill-operator-about.adoc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+[id="about-poison-pill-operator_{context}"]
+= About the Poison Pill Operator
+
+The Poison Pill Operator reboots unhealthy nodes in the cluster. The Operator uses the `MachineHealthCheck` controller to detect the health of a node in the cluster. 
+When a node is identified as unhealthy, the `MachineHealthCheck` resource creates the `PoisonPillRemediation` CR, which triggers the Poison Pill Operator. 
+
+The Poison Pill Operator provides the following capabilities:
+
+- Minimizes downtime for stateful applications and and restores compute capacity if transient failures occur.
+- Independent of any management interface such as IPMI or an API to provision a node.
```
Comment (ogradyp) [inline modules/eco-poison-pill-operator-about.adoc:14]: Include a comma before such as

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+If the `PoisonPillRemediation` CR is created, ensure that its name matches the unhealthy node or the machine object.
+
+[id="daemon-set-exists_{context}"]
+== Daemon set exists even after uninstalling the Poison Pill Operator 
+Issue:: The Poison Pill daemon set exists even after after uninstalling the Operator.
+
+Resolution:: To remove the Poison Pill daemon set, manually delete the `PoisonPillConfig` CR.
```
Comment (ogradyp) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:45]: For troubleshooting 'Unsuccessful remediation' you provide the command 'To do so, run the following command:' 
I think it would also be helpful to include the command for troubleshooting 'Daemon set exists even after uninstalling the Poison Pill Operator'.

File: _topic_map.yml [hunk 7]
```diff
@@ -1638,6 +1638,8 @@ Topics:
     File: adding-bare-metal-compute-user-infra
 - Name: Deploying machine health checks
   File: deploying-machine-health-checks
+- Name: Poison Pill Operator for remidiating unhealthy nodes
```
Comment (ogradyp) [inline _topic_map.yml:1641]: A question more than a comment. 
I've noticed 'Poison Pill Operator' is fully spelt out every time, is it ever / or can it be abbreviated to PPO?

Comment (abhatt-rh) [reply _topic_map.yml:1641]: @ogradyp We can't abbreviate it- has been spelled out in all engineering documents and PRs.

File: modules/eco-poison-pill-operator-about.adoc [hunk 8]
```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+// * /machine_management/eco-poison-pill-operator.adoc
+
+[id="about-poison-pill-operator_{context}"]
+= About the Poison Pill Operator
+
+The Poison Pill Operator reboots unhealthy nodes in the cluster. The Operator uses the `MachineHealthCheck` controller to detect the health of a node in the cluster. When a node is identified as unhealthy, the `MachineHealthCheck` resource creates the `PoisonPillRemediation` custom resource (CR), which triggers the Poison Pill Operator.
```
Comment (beekhof) [inline modules/eco-poison-pill-operator-about.adoc:8]:
```suggestion
The Poison Pill Operator runs on cluster nodes in order to reboot them when they become unhealthy. The Operator relies on the `MachineHealthCheck` controller to detect the health of a node in the cluster. When a node is identified as unhealthy, the `MachineHealthCheck` resource creates the `PoisonPillRemediation` custom resource (CR), which triggers the Poison Pill Operator. 
```

File: modules/eco-poison-pill-operator-about.adoc (hunk 1 above)
Comment (beekhof) [reply modules/eco-poison-pill-operator-about.adoc:51]:
```suggestion
<1> Specify the timeout duration, after which for surviving peer Operators can assume that an unhealthy node has been rebooted. The Operator automatically calculates the lower limit for this value. However, if different nodes have different watchdog timeouts, or you want to be especially cautious, you must change this value to a higher value.
```

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+
+[id="general-troubleshooting-poison-pill-operator_{context}"]
+== General Troubleshooting 
+
+Issue::
+You want to troubleshoot issues with the Poison Pill Operator.
+
+Resolution::
+Enable the debug flag in the `poison-pill` object. The `debug` flag increases
```
Comment (beekhof) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:16]: The example CR in "Understanding the Poison Pill Operator configuration" does not list a "debug" flag.

File: modules/eco-poison-pill-operator-troubleshooting.adoc (hunk 4 above)
Comment (beekhof) [reply modules/eco-poison-pill-operator-troubleshooting.adoc:37]: The `MachineHealthCheck` controller

**Review by beekhof — CHANGES_REQUESTED** [review]

File: modules/eco-poison-pill-operator-troubleshooting.adoc (hunk 9 above)
Comment (abhatt-rh) [reply modules/eco-poison-pill-operator-troubleshooting.adoc:16]: Updated the resolution section to check pod logs.

File: modules/eco-poison-pill-operator-troubleshooting.adoc (hunk 6 above)
Comment (abhatt-rh) [reply modules/eco-poison-pill-operator-troubleshooting.adoc:45]: Thank you; added the command

**Comment by abhatt-rh** [comment]
Thank you @ogradyp @johnwilkins, and @beekhof for your inputs and suggestions. I have updated the content accordingly.

File: modules/eco-poison-pill-operator-installation-web-console.adoc [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+//
+// * /machine_management/eco-poison-pill-operator.adoc
+
+[id="installing-poison-pill-operator-using-web-console_{context}"]
+= Installing the Poison Pill Operator using the web console
+
+.Prerequisites
+
+* A cluster installed on bare-metal hardware using installer-provisioned infrastructure (IPI).
+* Install the OpenShift CLI (`oc`).
```
Comment (sagidlow) [inline modules/eco-poison-pill-operator-installation-web-console.adoc:11]: Would the user need access to the CLI if they are using the Web Console? Or would they just need access to the Web Console?

**Review by sagidlow — COMMENTED** [review]
Really great work on this PR. I left some comments, let me know if you have any questions.

File: modules/eco-poison-pill-operator-installation-web-console.adoc (hunk 10 above)
Comment (abhatt-rh) [reply modules/eco-poison-pill-operator-installation-web-console.adoc:11]: Corrected - they would need the web console

**Comment by abhatt-rh** [comment]
@sagidlow Thank you for the inputs and suggestions. I have incorporated them and updated the content. PTAL. Thank you!

**Comment by abhatt-rh** [comment]
https://github.com/openshift/openshift-docs/pull/35974#issuecomment-913231453 
@n1r1 @shellymiron Requesting your review and acks. I have addressed Andrew's feedback and other peer review comments

File: modules/eco-poison-pill-operator-about.adoc [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    - apiVersion: poison-pill.medik8s.io/v1alpha1
+      fieldsType: FieldsV1
+      fieldsV1:
+        'f:spec':
+          .: {}
+          'f:safeTimeToAssumeNodeRebootedSeconds': {}
+          'f:watchdogFilePath': {}
+      manager: manager
+      operation: Update
+      time: '2021-08-23T22:20:49Z'
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-about.adoc:41]: I think we can omit these lines as they don't add a real value for the users

File: modules/eco-poison-pill-operator-about.adoc [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+  name: poison-pill-config
+  namespace: openshift-operators
+  resourceVersion: '31617'
+  uid: 7c9c4502-f847-4deb-ad7d-877551777a49
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-about.adoc:45]: same for these

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+----
+
+. Verify that the Poison Pill Operator created the `PoisonPillConfig` CR:
++
+[source,terminal]
+----
+$ oc get PoisonPillConfig -n poison-pill
+----
+
+. Verify that each poison pill pod is scheduled and running on each node:
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-installation-cli.adoc:99]: should be on each *worker* node, as we don't run on masters

File: modules/eco-poison-pill-operator-installation-web-console.adoc [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+// *nodes/nodes/eco-poison-pill-operator.adoc
+
+[id="installing-poison-pill-operator-using-web-console_{context}"]
+= Installing the Poison Pill Operator using the web console
+
+The following procedure installs the Poison Pill Operator using the {product-title} web console.
+
+.Prerequisites
+
+* A cluster installed on bare-metal hardware using installer-provisioned infrastructure (IPI).
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-installation-web-console.adoc:12]: Poison Pill can work on IPI and also on non-baremetal clusters so I think we can omit this

File: modules/eco-poison-pill-operator-installation-web-console.adoc [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-installation-web-console.adoc:15]: We should mention something about modifying the MHC CR to trigger Poison Pill. not sure where's the right place for that

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+[id="unsuccessful_remediation{context}"]
+== Unsuccessful remediation 
+Issue:: An unhealthy node was not remediated.
+
+Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. Run the following command:
++
+[source,terminal]
+----
+$ oc get namespace <machinehealthcheck>
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:32]: should be `oc get ppr -A`

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+[id="daemon-set-exists_{context}"]
+== Daemon set exists even after uninstalling the Poison Pill Operator 
+Issue:: The Poison Pill daemon set exists even after after uninstalling the Operator.
+
+Resolution:: To remove the Poison Pill daemon set, manually delete the `PoisonPillConfig` CR. Run the following command:  
++
+[source,terminal]
+----
+$ oc delete pods <poison-pill-daemonset> -n <namespace>
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:47]: should be `oc delete ds ...`

**Review by n1r1 — COMMENTED** [review]
looks good!
added some comments. 
I'm mainly missing the part of the MHC CR configuration that triggers the Poison Pill.

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+Issue:: An unhealthy node was not remediated.
+
+Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. Run the following command:
++
+[source,terminal]
+----
+$ oc get namespace <machinehealthcheck>
+----
++
+If the `MachineHealthCheck` controller did not create the `PoisonPillRemediation` CR when the node turned unhealthy, check the logs of the `MachineHealthCheck` controller.
```
Comment (shellymiron) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:35]: maybe it is worth mentioning that a PoisonPillRemediationTemplate resource needs to exists (to be created manually)
in the cluster, for the machine health check to create the ppr

File: modules/eco-poison-pill-operator-installation-web-console.adoc (hunk 15 above)
Comment (abhatt-rh) [reply modules/eco-poison-pill-operator-installation-web-console.adoc:14]: Added a new section (procedure) to address this

**Comment by abhatt-rh** [comment]
@n1r1 @shellymiron Thank you for the quick review. I have incorporated your comments updated the content. PTAL

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+= Installing the Poison Pill Operator using the CLI
+
+The following procedure installs the Poison Pill Operator using the CLI.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-installation-cli.adoc:15]: For this procedure, I recommend matching the convention that is used in the next procedure ("Configuring machine health checks") where you first define the file and show the example YAML and then you use the `oc create` command to create the resource. This is generally how we instruct users in most of our docs, AFAIK. 

I'm thinking it would be more conventional if it were something like this:
1. Define a namespace:
	_Example `namespace.yaml`_ block header and YAML block
2. Create the namespace:
	`$ oc create -f namespace.yaml`

Here's an example of this in our docs: https://docs.openshift.com/container-platform/4.8/updating/installing-update-service.html#update-service-install-cli_update-service

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 20]
```diff
@@ -0,0 +1,109 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes/eco-poison-pill-operator.adoc
+
+[id="installing-poison-pill-operator-using-cli_{context}"]
+= Installing the Poison Pill Operator using the CLI
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-installation-cli.adoc:6]: s/using/by using/ to avoid ambiguity here and throughout (per IBM Style Guide)

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+.Procedure
+
+. To create a `Namespace` object YAML file, run the following command:
++
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
+----
++
+.Sample YAML file
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-installation-cli.adoc:24]: In addition to my previous comment, I'd suggest giving this sample a name, such as `poison-pill.yaml` and then using that in the commands and definitions throughout the procedure.

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 22]
```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes/eco-poison-pill-operator.adoc
+
+[id="troubleshooting-poison-pill-operator_{context}"]
+= Troubleshooting the Poison Pill Operator
+
+
+[id="general-troubleshooting-poison-pill-operator_{context}"]
+== General Troubleshooting
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:10]: This should be sentence case ("General troubleshooting")

**Review by bobfuru — COMMENTED** [review]
This looks fantastic, @abhatt-rh! I left a few suggestions for your consideration. Most importantly, I recommend revising the "Installing the Poison Pill Operator" procedure to align with the layout of your other procedures, where users first define the resource, then create it. I left more details in some notes. Let me know if you need any clarification on my comments. Thanks!

**Comment by abhatt-rh** [comment]
@bobfuru Thank you for review. I have incorporated your suggestions. PTAL. Thank you!

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+----
+.. To create the `Namespace` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f poison-pill-namespace.yaml
+----
+
+. Create an `OperatorGroup` CR:
+.. Create the following `OperatorGroup` CR and save the YAML file, for example, `poison-pill-operator-group.yaml`:
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-installation-cli.adoc:35]: You're not actually creating the CR until step 1b, so I'd rephrase this slightly:

```suggestion
.. Define the `OperatorGroup` CR and save the YAML file, for example, `poison-pill-operator-group.yaml`:
```

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+----
+.. To create the `OperatorGroup` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f poison-pill-operator-group.yaml
+----
+
+. Create a `Subscription` CR:
+.. Create the following `Subscription` CR and save the YAML file, for example, `poison-pill-subscription.yaml`:
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-installation-cli.adoc:56]: Same as my previous comment:
```suggestion
.. Define the `Subscription` CR and save the YAML file, for example, `poison-pill-subscription.yaml`:
```

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 25]
```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes/eco-poison-pill-operator.adoc
+
+[id="troubleshooting-poison-pill-operator_{context}"]
+= Troubleshooting the Poison Pill Operator
+
+
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:8]: Please remove the extra line break

File: modules/eco-poison-pill-operator-about.adoc [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+* Independent of any management interface, such as IPMI or an API to provision a node. 
+
+[id="understanding-poison-pill-operator-config_{context}"]
+== Understanding the Poison Pill Operator configuration
+
+The Poison Pill Operator creates the `PoisonPillConfig` CR with the name `poison-pill-config` in the Poison Pill Operator's namespace. You can edit this CR. However, you cannot create a new CR for the Poison Pill Operator.
+
+A change in the `PoisonPillConfig` CR re-creates the Poison Pill daemon set.
+
+The `PoisonPillConfig` CR resembles the following YAML file:
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-about.adoc:22]:
```suggestion
The `PoisonPillConfig` definition resembles the following YAML file:
```

**Review by bobfuru — COMMENTED** [review]
A few minor requests for clarification in the wording but overall this LGTM.

File: modules/eco-poison-pill-operator-about.adoc (hunk 26 above)
Comment (abhatt-rh) [reply modules/eco-poison-pill-operator-about.adoc:22]: > 
Retaining `PoisonPillConfig` CR per our discussion
Addressed the remaining comments. Thank again!

File: modules/eco-configuring-machine-health-check-with-poison-pill.adoc [hunk 27]
```diff
@@ -0,0 +1,76 @@
...
+Use the following procedure to configure the machine health checks to use the Poison Pill Operator as a remediation provider.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a `PoisonPillRemediationTemplate` CR
```
Comment (bobfuru) [inline modules/eco-configuring-machine-health-check-with-poison-pill.adoc:17]: This should have a colon at the end of the sentence to match elsewhere in the assembly.

File: modules/eco-configuring-machine-health-check-with-poison-pill.adoc [hunk 28]
```diff
@@ -0,0 +1,76 @@
...
+    apiVersion: poison-pill.medik8s.io/v1alpha1
+    name: <poisonpillremediationtemplate-sample>
+----
+<1> Specify the details for the remediation template.
++
+.. To create a `MachineHealthCheck` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f <machinehealthcheck>.yaml
```
Comment (bobfuru) [inline modules/eco-configuring-machine-health-check-with-poison-pill.adoc:67]: Most commonly, we write sample filenames as such:
`$ oc create -f <file-name>.yaml`
If you wanted, you could annotate this and note in the callout something like, `<1> The name of the machine health check file.`

File: modules/eco-configuring-machine-health-check-with-poison-pill.adoc [hunk 29]
```diff
@@ -0,0 +1,76 @@
...
+----
+.. To update a `MachineHealthCheck` CR, run the following command:
++
+[source,terminal]
+----
+$ oc apply -f <mhc-update>.yaml
```
Comment (bobfuru) [inline modules/eco-configuring-machine-health-check-with-poison-pill.adoc:73]: Same as my previous comment about `<file-name>.yaml`

File: modules/eco-configuring-machine-health-check-with-poison-pill.adoc [hunk 30]
```diff
@@ -0,0 +1,76 @@
...
+spec:
+  template:
+    spec: {}
+----
+
+.. To create the `PoisonPillRemediationTemplate` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f <ppr-name>.yaml
```
Comment (bobfuru) [inline modules/eco-configuring-machine-health-check-with-poison-pill.adoc:37]: See my later comment about a more common usage in our doc set: `<file-name>.yaml`

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+== Checking the daemon set 
+Issue:: The Poison Pill Operator is installed but the daemon set is not available.
+
+Resolution:: Check the Operator logs for errors or warnings.
+
+[id="unsuccessful_remediation{context}"]
+== Unsuccessful remediation 
+Issue:: An unhealthy node was not remediated.
+
+Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. Run the following command:
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:27]: Consider a small revision for clarity:
```suggestion
Resolution:: Verify that the `PoisonPillRemediation` CR was created by running the following command:
```

File: modules/eco-poison-pill-operator-troubleshooting.adoc [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
++
+[source,terminal]
+----
+$ oc get ppr -A
+----
++
+If the `MachineHealthCheck` controller did not create the `PoisonPillRemediation` CR when the node turned unhealthy, check the logs of the `MachineHealthCheck` controller. Additionally, ensure that the `MachineHealthCheck` CR includes the required specification to use the remediation template.
++
+If the `PoisonPillRemediation` CR is created, ensure that its name matches the unhealthy node or the machine object.
```
Comment (bobfuru) [inline modules/eco-poison-pill-operator-troubleshooting.adoc:36]: s/is created/was created/ (because it already happened)

**Review by bobfuru — COMMENTED** [review]
A few more comments for consideration. I apologize for not catching these before. Let me know when you're ready for a merge and cherrypick. Thanks!

File: modules/eco-poison-pill-operator-installation-cli.adoc [hunk 33]
```diff
@@ -0,0 +1,112 @@
...
+.. To create the `Subscription` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f poison-pill-subscription.yaml
+----
+
+.Verification
+
+. Verify the installation succeeded by inspecting the CSV file:
```
Comment (n1r1) [inline modules/eco-poison-pill-operator-installation-cli.adoc:81]: s/file/resource

File: modules/eco-configuring-machine-health-check-with-poison-pill.adoc [hunk 34]
```diff
@@ -0,0 +1,76 @@
...
+----
+
+. Create or update the `MachineHealthCheck` CR to  point to the `PoisonPillRemediationTemplate` CR.
```
Comment (n1r1) [inline modules/eco-configuring-machine-health-check-with-poison-pill.adoc:40]: nit: double space between "to" and "point"

File: modules/eco-configuring-machine-health-check-with-poison-pill.adoc [hunk 35]
```diff
@@ -0,0 +1,76 @@
...
+  namespace: openshift-machine-api
+spec:
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-machine-role: "worker"
+      machine.openshift.io/cluster-api-machine-type: "worker"
+  remediationTemplate: <1>
+    kind: PoisonPillRemediationTemplate
+    apiVersion: poison-pill.medik8s.io/v1alpha1
+    name: <poisonpillremediationtemplate-sample>
```
Comment (n1r1) [inline modules/eco-configuring-machine-health-check-with-poison-pill.adoc:59]: I think we should keep the `unhealthyConditions` section of MHC here, to provide a complete example

**Comment by bobfuru** [comment]
LGTM
